src/Notify/Notifier/AbstractFcmNotifier.php

```diff
@@ -26,15 +26,18 @@ abstract class AbstractFcmNotifier implements Notifier
     const API_ENDPOINT = 'https://fcm.googleapis.com';
     const DISABLED_FLAG_NAME = 'fcm_notifier_disabled';
 
-    protected string $apiKey = '';
     protected Flags $flags;
     protected ?GuzzleClient $guzzleClient = null;
     protected Key $key;
+    protected string $accessToken = '';
+    protected array $config = [];
+    protected string $firebaseServiceAccountSecrets = '';
```

Comment:
@bhsiao55 shouldn't this be app specific, i.e. coming from the provided app object?

Reply:
@gdbrown not necessarily, we have a single firebase project per env `acme-env` that is configured to use a service account for fcm - notifications can be customized per platform:

> Whenever you want to send values only to particular platforms, don't use common fields; use platform-specific fields. For example, to send a notification only to Apple platforms and web but not to Android, you must use two separate sets of fields, one for Apple and one for web.
- https://firebase.google.com/docs/cloud-messaging/send-message#customize-messages-across-platforms

Reply:
(we won't be using the ios/android apps at all once we are using the v1 api)

Reply:
looking closer, think the apps have always used the exact same values for fcm_api_key anyways

Reply:
@gdbrown i could see it either way.. its also odd to set up an android/ios/browser-app that is capable of notifying all three. what we're really talking about here is an fcm-app

Reply:
@gdbrown the firebaseServiceAccountSecrets is used to obtain the Oauth access token, this token that is generated can be used for any of the the 3 apps we have, Android, Ios, and Browser. The payload we send, determines which platform the message is being sent to.  https://firebase.google.com/docs/cloud-messaging/migrate-v1#provide-credentials-manually

Reply:
@efjacobson @bhsiao55 this new strategy is a departure, though does one imagine you'd need to control when/if platforms get sent a notif?  like in case X, only send to android, etc.

also this changes the automatic notif creation when the app is resolved in the notifications watcher.

Reply:
@gdbrown  im not actually suggesting we make and start using an fcm-app or fcm-notifications, just saying that these things dont fit neatly in to one another and its a bit of a rabbit hole. triniti apps are kind of service accounts but not really (they dont act for the api, they provide config to the api), and they kind of represent actual apps but not really (why is there an apple news app but not a jwplayer app?)

where would you say the line is drawn? if each concrete fcm notifier were injected with its "own" key, would that be separation enough? or is the ability to update at runtime the juice to this squeeze?